modules/core/js/src/main/scala/crystal/react/hooks/UseSingleEffect.scala

```diff
@@ -17,44 +17,52 @@ import scala.concurrent.duration.FiniteDuration
 
 class UseSingleEffect[F[_]](
   latch:    Ref[F, Option[Deferred[F, UnitFiber[F]]]],
+  cleanup:  Ref[F, Option[F[Unit]]],
   debounce: Option[FiniteDuration]
 )(using F: Async[F], monoid: Monoid[F[Unit]]) {
   private val debounceEffect: F[Unit] = debounce.map(F.sleep).orEmpty
 
-  private def switchTo(effect: F[Unit]): F[Unit] =
+  private def switchTo(effect: F[F[Unit]]): F[Unit] =
     Deferred[F, UnitFiber[F]] >>= (newLatch =>
       latch
-        .modify(oldLatch =>
-          (newLatch.some,
-           for {
-             // Cleanup latch after effect + debounce, so that we don't run debounce again next time.
-             newFiber <- (oldLatch.map(_.get.flatMap(_.cancel >> debounceEffect)).orEmpty >>
-                           effect >> debounceEffect >> latch.set(none)).start
-             _        <- newLatch.complete(newFiber)
-           } yield ()
+        .modify: oldLatch =>
+          (
+            newLatch.some,
+            (
+              oldLatch
```

Comment:
I feel like all these effects are a little hard to read. Could you separate some of them out into vals and chain those? That way it's a little easier to read what happens

Reply:
That's a good idea, I'll send another commit.